Time: 2025-05-12T13:25:21.535756700
Exception message: Could not register item due to system error
se.kth.iv1350.processSale.controller.ItemRegistrationFailureException: Could not register item due to system error
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:79)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:57)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:32)
Caused by: se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:69)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:43)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:65)
	... 2 more
--------------------------------------------------
Date: 2025-05-12
Time: 13:26:41.286990600
Exception message: Could not register item due to system error
se.kth.iv1350.processSale.controller.ItemRegistrationFailureException: Could not register item due to system error
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:79)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:57)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:32)
Caused by: se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:69)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:43)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:65)
	... 2 more
--------------------------------------------------
Date: 2025-05-12
Time: 15:23:31.427941900
Exception message: Could not register item due to system error
se.kth.iv1350.processSale.controller.ItemRegistrationFailureException: Could not register item due to system error
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:78)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:73)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:33)
Caused by: se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:70)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:44)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:72)
	... 2 more
--------------------------------------------------
Date: 2025-05-13
Time: 16:34:53.186134400
Exception message: Could not register item due to system error
se.kth.iv1350.processSale.controller.ItemRegistrationFailureException: Could not register item due to system error
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:79)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:73)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:33)
Caused by: se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:73)
	... 2 more
--------------------------------------------------
Date: 2025-05-16
Time: 13:50:47.999117800
Exception message: Could not register item due to system error
se.kth.iv1350.processSale.controller.ItemRegistrationFailureException: Could not register item due to system error
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:79)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:73)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:33)
Caused by: se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:73)
	... 2 more
--------------------------------------------------
Date: 2025-05-16
Time: 14:19:52.187498400
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:21:09.222838700
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:23:48.899918400
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:24:34.259710100
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:28:09.859696600
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:29:14.104505800
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:31:15.507320200
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:32:46.979526400
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:34:52.297534700
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:71)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:37:00.505147100
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:38:18.373944600
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:43:17.196180600
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-16
Time: 14:56:55.934542500
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.controller.ControllerTest.lambda$0(ControllerTest.java:57)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:39)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3153)
	at se.kth.iv1350.processSale.controller.ControllerTest.testIfRegisterNewItemThrowsItemRegistrationFailureException(ControllerTest.java:55)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:775)
	at org.junit.platform.commons.support.ReflectionSupport.invokeMethod(ReflectionSupport.java:479)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:161)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:152)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:91)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:112)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:94)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:87)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$4(TestMethodTestDescriptor.java:220)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:216)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:158)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:70)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:157)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.executeEngine(EngineExecutionOrchestrator.java:215)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.failOrExecuteEngine(EngineExecutionOrchestrator.java:195)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:164)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:95)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:60)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:143)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:59)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:94)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:52)
	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$execute$2(InterceptingLauncher.java:47)
	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
	at org.junit.platform.launcher.core.InterceptingLauncher.execute(InterceptingLauncher.java:46)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:52)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:70)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestReference.run(JUnit5TestReference.java:100)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:520)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:748)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:443)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:211)
--------------------------------------------------
Date: 2025-05-16
Time: 15:01:20.107183400
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:76)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:70)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-17
Time: 15:54:21.170060700
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:77)
	at se.kth.iv1350.processSale.view.View.runFakeExecution(View.java:69)
	at se.kth.iv1350.processSale.startup.Main.main(Main.java:34)
--------------------------------------------------
Date: 2025-05-17
Time: 16:49:30.768025200
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:77)
	at se.kth.iv1350.processSale.controller.ControllerTest.lambda$0(ControllerTest.java:57)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:39)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3153)
	at se.kth.iv1350.processSale.controller.ControllerTest.testIfRegisterNewItemThrowsItemRegistrationFailureException(ControllerTest.java:55)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:775)
	at org.junit.platform.commons.support.ReflectionSupport.invokeMethod(ReflectionSupport.java:479)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:161)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:152)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:91)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:112)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:94)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:87)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$4(TestMethodTestDescriptor.java:220)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:216)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:158)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:70)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:157)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.executeEngine(EngineExecutionOrchestrator.java:215)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.failOrExecuteEngine(EngineExecutionOrchestrator.java:195)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:164)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:95)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:60)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:143)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:59)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:94)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:52)
	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$execute$2(InterceptingLauncher.java:47)
	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
	at org.junit.platform.launcher.core.InterceptingLauncher.execute(InterceptingLauncher.java:46)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:52)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:70)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestReference.run(JUnit5TestReference.java:100)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:520)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:748)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:443)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:211)
--------------------------------------------------
Date: 2025-05-17
Time: 16:52:46.994074600
Exception message: Database failure occurred while searching for item with ID: 7891
se.kth.iv1350.processSale.integration.DataBaseFailureException: Database failure occurred while searching for item with ID: 7891
	at se.kth.iv1350.processSale.integration.InventorySystem.simulateDataBaseFailure(InventorySystem.java:71)
	at se.kth.iv1350.processSale.integration.InventorySystem.validateIdentifier(InventorySystem.java:45)
	at se.kth.iv1350.processSale.controller.Controller.registerNewItem(Controller.java:77)
	at se.kth.iv1350.processSale.controller.ControllerTest.lambda$0(ControllerTest.java:57)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:53)
	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:39)
	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3153)
	at se.kth.iv1350.processSale.controller.ControllerTest.testIfRegisterNewItemThrowsItemRegistrationFailureException(ControllerTest.java:55)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:775)
	at org.junit.platform.commons.support.ReflectionSupport.invokeMethod(ReflectionSupport.java:479)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:161)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:152)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:91)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:112)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:94)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:87)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$4(TestMethodTestDescriptor.java:220)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:216)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:158)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:70)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:157)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:161)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:147)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:145)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:144)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:101)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.executeEngine(EngineExecutionOrchestrator.java:215)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.failOrExecuteEngine(EngineExecutionOrchestrator.java:195)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:164)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:95)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:60)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:143)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:59)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:94)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:52)
	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$execute$2(InterceptingLauncher.java:47)
	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
	at org.junit.platform.launcher.core.InterceptingLauncher.execute(InterceptingLauncher.java:46)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:52)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:70)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestReference.run(JUnit5TestReference.java:100)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:520)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:748)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:443)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:211)
--------------------------------------------------
